.circleci/config.yml

```diff
@@ -0,0 +1,174 @@
+     steps: 
+        - checkout
+        - run: 
+           name: Checkout Submodules
+           command: |
+              git submodule sync
+              git submodule update --init --recursive --jobs 8
+        - run: 
+           command: brew update
+        - run: 
+           command: brew install coreutils 
+        - run:
+           command: |
+              brew install go  ccache gflags &&
+              brew install libscrypt libtool cmake libjson-rpc-cpp conan
+        - restore_cache:
+           keys:
+            - conan-cache-v1-mac--{{ checksum "conanfile.py" }}
+            - conan-cache-v1-mac
+        - run:
+           name: connan install 
+           command: conan install -if cmake-build --build missing .
+        - save_cache:
+           paths:
+             - ~/.conan
+           key: conan-cache-v1-mac--{{ checksum "conanfile.py" }}
+        - run: 
+           name: conan Build
+           command: |
+             export GOPATH=$HOME/golang &&
+             export GOROOT=/usr/local/opt/go/libexec &&
+             export CC=gcc &&
+             export CPATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include &&
+             conan build -bf cmake-build -sf . .  &&
+             cp cmake-build/bin/taraxad cmake-build/bin/taraxad-macos
+        - store_artifacts:
+           path: /Users/distiller/project/cmake-build/bin/taraxad
+        - persist_to_workspace:
+           root: cmake-build
+           paths: [ bin ]
+   build-linux:
+     environment:
+        BUILD_OUTPUT_DIR: "cmake-docker-build-debug"
+        CLANG_VERSION: "1:10.0-50~exp1"
+        GO_VERSION: "1.13.7"
+        CMAKE_VERSION: "3.16.3-1ubuntu1"
+        GCC_VERSION: "4:9.3.0-1ubuntu2"
+        GFLAGS_VERSION: "2.2.2-1build1"
+        JSONCPP_VERSION: "1.7.4-3.1ubuntu2"
+        JSONRPCCPP_VERSION: "0.7.0-1build3"
+        SCRYPT_VERSION: "1.21-3"
+        GOROOT: "/usr/local/go"
+        GOPATH: "/root/.go"
+     docker:
+      - image: gcr.io/$GOOGLE_PROJECT_ID/taraxa-node/builder
+        auth:
+          username: _json_key  # default username when using a JSON key file to authenticate
+          password: $GCLOUD_SERVICE_KEY 
+     resource_class: xlarge
+     steps:
+        - checkout
+        - run:       
+           name: Checkout Submodules
+           command: |
+              git submodule sync
+              git submodule update --init --recursive --jobs 8
+        - run: 
+           command: mkdir $BUILD_OUTPUT_DIR
+        - run: 
+           name: Copy /opt/taraxa/existing files to project
+           command: cp -rp /opt/taraxa/cmake-docker-build-debug/* /root/project/cmake-docker-build-debug/
+#       - git/checkout-with-submodules
+#       - setup_remote_docker:
+#           docker_layer_caching: true
+        - run: 
+            name: Cmake
+            command: | 
+              cd $BUILD_OUTPUT_DIR &&
+              cmake -DCMAKE_BUILD_TYPE=Debug \
```

Comment:
We will probably want this to be `Release` instead of `Debug` and we will need to work with conan as well